.github/workflows/automerge-for-humans-merging.yml

```
@@ -18,38 +14,88 @@ on:
-          cmd: | 
-            curl -H "Accept: application/vnd.github+json" -H "Authorization: Bearer ${{ secrets.GH_TOKEN }}" "${{github.event.pull_request._links.commits.href}}?per_page=100" | 
-              jq -r '[.[] 
-                | {name: .commit.author.name, email: .commit.author.email, login: .author.login}] 
-                | map(select(.login != "${{github.event.pull_request.user.login}}")) 
-                | unique 
-                | map("Co-authored-by: " + .name + " <" + .email + ">") 
-                | join("\n")'
-          multiline: true
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const { Octokit } = require('@octokit/rest');
+            const { paginateRest } = require('@octokit/plugin-paginate-rest');
+
+            const fetch = require('cross-fetch');
+
+            const token = process.env.GITHUB_TOKEN;
+            const prNumber = process.env.PR_NUMBER;
+            const repository = process.env.GITHUB_REPOSITORY;
+
+            async function getCoAuthors() {
+              try {
+                const octokit = new Octokit({
+                  auth: token,
+                  request: {
+                    fetch,
+                  },
+                });
+                  
+                const commitsResponse = await octokit.paginate("GET /repos/{owner}/{repo}/pulls/{pull_number}/commits", {
+                  owner: repository.split('/')[0],
+                  repo: repository.split('/')[1],
+                  pull_number: ${{ github.event.number }},
+                  per_page: 100,
+                });
+                  
+                const authors = commitsResponse
+                  .map(data => ({
+                    name: data.commit.author.name,
+                    email: data.commit.author.email,
+                    login: data.commit.author.login,
+                  }))
+                  .filter(author => author.login !== 'PR_sender_login')
+                  .reduce((uniqueAuthors, author) => {
+                    if (!uniqueAuthors.some(a => a.email === author.email)) {
+                      uniqueAuthors.push(author);
+                    }
+                    return uniqueAuthors;
+                  }, [])
+                  .map(author => `Co-authored-by: ${author.name} <${author.email}>`)
+                  .join('\n');
+                console.log(authors);
+                return authors;
+              } catch (error) {
+                console.error('Error fetching commits:', error);
+                return null;
+              }
+            }
+          
+            await getCoAuthors();
+
       - name: Automerge PR
-        uses: pascalgn/automerge-action@22948e0bc22f0aa673800da838595a3e7347e584 #v0.15.6 https://github.com/pascalgn/automerge-action/releases/tag/v0.15.6
+        uses: pascalgn/automerge-action@v0.15.6
```

```suggestion
        uses: pascalgn/automerge-action@22948e0bc22f0aa673800da838595a3e7347e584 #v0.15.6 https://github.com/pascalgn/automerge-action/releases/tag/v0.15.6
```

we use the commit instead of release tag for security reasons.